### PR TITLE
ADD Deployment Guide

### LINKED ISSUE #151: [Feature Request]:  ADD Deployment Guide
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

Adding a Deployment Guide to your README is essential for several reasons. It facilitates easy setup, allowing users and contributors to deploy the application in their environments without confusion quickly. This clarity encourages adoption, as users are more likely to try out an app they can easily run. A standardized guide ensures that everyone follows the same deployment practices, reducing the likelihood of issues stemming from different configurations and making troubleshooting more straightforward. Additionally, it enhances community contributions by enabling contributors to test their changes locally before submitting pull requests. The guide can document best practices for security and environment configuration, providing valuable insights for users deploying the app in production. By clarifying prerequisites and dependencies, the deployment guide saves users time and frustration, making the project more accessible. Overall, a well-crafted deployment guide not only improves user experience but also enhances the visibility and attractiveness of your project within the open-source community.

### Record

- [X] I agree to follow this project's Code of Conduct
- [X] I want to work on this issue
- [X] I'm willing to provide further clarification or assistance if needed.

### PR DESCRIPTION
**Pull request by alo7lika**
- Fixes #151

## 📋 Description
Added a comprehensive Deployment Guide for the Full Stack Chatting App to assist users in deploying the application on their servers.

## 🔨 Changes Made

- Created a new section in the README for the Deployment Guide.
- Provided detailed steps for deploying both frontend and backend components.
- Included configurations for web servers like NGINX for handling routing and SSL setup.

## 🔨 Screen Shots OR Short videos of working
Screenshot 2024-10-24 171245

## ✅ Checklist

Before submitting the PR, please make sure you have completed the following:

- [x] My code follows the style guidelines of this project.
- [x] Read Contributing.md

## 🏷️ Types of Changes

What type of changes does your code introduce? (Check all that apply)

- [ ] Bug fix (non-breaking change which fixes an issue) 🐛
- [ ] New feature (non-breaking change which adds functionality) ✨
- [ ] UI enhancement (non-breaking change which enhances UI) 🎨
- [x] Documentation update 📚

## 🤝 Related Issues
N/A

## Reminder:
Please ensure your pull request is made to the `test` branch, not `main`.

*Thank you for contributing to Talk-Trove! We look forward to reviewing your PR and getting it merged!*

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,9 +123,114 @@ yarn seed
...
+### 2. Configure Environment Variables 📋
+
+- Create a `.env` file in the backend directory.
+- Make sure to include the following variables:
+  
+  ```plaintext
+  MONGO_URI=your_mongodb_connection_string
+  JWT_SECRET=your_jwt_secret
+  NODE_ENV=production
+  PORT=your_preferred_port
```
Comment (Sibam-Paul) [inline README.md:148]: Mention there is a env.sample in the file to rename it to .env just tell all the above are there in .env.sample file and mention the above and just rename it as instructions in the readme

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -123,9 +123,114 @@ yarn seed
...
+
+- Navigate to the `backend` directory on your server.
+- Install dependencies:
+
+  ```bash
+  yarn install
+  ```
+### 5. Deploying Client App 📱
+
+- Use a web server like **NGINX** or **Apache** to serve the frontend.
```
Comment (Sibam-Paul) [inline README.md:173]: Also add if u don't have mongodb write the setup of docker

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]

**Comment by alo7lika** [comment]
@kartikayasijaa  and @Sibam-Paul  the task has been completed.
Kindly add the labels GSSOC Ext 24 and HACKTOBERFEST

**Comment by alo7lika** [comment]
> @kartikayasijaa and @Sibam-Paul the task has been completed. Kindly add the labels GSSOC Ext 24 and HACKTOBERFEST

Made the changes @Sibam-Paul

**Comment by alo7lika** [comment]
> > @kartikayasijaa and @Sibam-Paul the task has been completed. Kindly add the labels GSSOC Ext 24 and HACKTOBERFEST
> 
> Made the changes @Sibam-Paul

@Sibam-Paul

**Comment by kartikayasijaa** [comment]
@alo7lika Can we connect over discord for this PR?

**Comment by alo7lika** [comment]
> @alo7lika Can we connect over discord for this PR?

Do I need to make any more changes?

**Comment by kartikayasijaa** [comment]
Yes, we can't add this in readme. This information will not make any sense there.
@alo7lika

**Pull request closed without merging**